config/opal_check_rocm.m4

```diff
@@ -0,0 +1,72 @@
+dnl
+dnl Copyright (C) 2022      Advanced Micro Devices, Inc. All rights reserved.
+dnl $COPYRIGHT$
+dnl
+dnl Additional copyrights may follow
+dnl
+dnl $HEADER$
+dnl
+
+
+# OMPI_CHECK_ROCM(prefix, [action-if-found], [action-if-not-found])
+# --------------------------------------------------------
+# check if ROCM support can be found.  sets prefix_{CPPFLAGS,
+# LDFLAGS, LIBS} as needed and runs action-if-found if there is
+# support, otherwise executes action-if-not-found
+
+
+#
+# Check for ROCm  support
+#
+AC_DEFUN([OPAL_CHECK_ROCM],[
+
+     OPAL_VAR_SCOPE_PUSH([opal_check_rocm_happy rocm_save_CPPFLAGS rocm_save_LDFLAGS])
+
+     rocm_save_CPPFLAGS="$CPPFLAGS"
+     rocm_save_LDFLAGS="$LDFLAGS"
+     
+     # Get some configuration information
+     AC_ARG_WITH([rocm],
+        [AS_HELP_STRING([--with-rocm(=DIR)],
+        [Build ROCm support, optionally adding DIR/include, DIR/lib, and DIR/lib64 to the search path for headers and libraries])])
+
+
+     CPPFLAGS="$CPPFLAGS -D__HIP_PLATFORM_AMD__ -I$with_rocm/include/hip -I$with_rocm/include"			   
```

Comment:
that's a good point. In fact, I should probably try to set a default directory in that case, since rocm is most of the time installed in /opt/rocm, so that would simply the users life by not having to specify the directory.